### ISSUE TITLE
Less specific server header in responses

### ISSUE THREAD

**Issue by mmzeeman** [zotonic/zotonic#1821]
One of our clients requested if we could remove the very specific server header zotonic currently has with a less specific one.

Currently the version number.

```
Server: MochiWeb/1.1 WebZMachine/2.0 (Z) Zotonic/0.31.0
```

And replace it with this.

```
Server: Zotonic
```

**Comment by mworrell** [comment]
I propose to change this for the 1.0.

As a workaround for the 0.x you can try to add the following line in your site code:

```
application:set_env(webzmachine, server_header, "Zotonic").
```

This must be called after the initialization of mochiweb/webmachine by Zotonic.
The schema install code might be a good place.

**Comment by mmzeeman** [comment]
:+1:

**Comment by CyBeRoni** [comment]
Interestingly I have a client with the same request. I'd then prefer to remove the header entirely; I assume the above is global. If you set it to "", does the header disappear?

**Comment by mworrell** [comment]
I think it will just be an empty header, so still present.

Cowboy sets it hard coded to `Cowboy`:

```Erlang
	Headers = maps:merge(#{
		<<"date">> => cowboy_clock:rfc1123(),
		<<"server">> => <<"Cowboy">>
	}, maps:merge(RespHeaders, Headers0)),
```

So we need to come up with some default value ourselves.

**Comment by mworrell** [comment]
In Zotonic we overrule this in one single place, in `zotonic_listen_http`

```Erlang
%% @doc Start the HTTP/S listeners
-spec start_http_listeners() -> ok.
start_http_listeners() ->
    z_ssl_certs:ensure_dhfile(),
    application:set_env(cowmachine, server_header,
        <<"Zotonic/", (z_convert:to_binary(?ZOTONIC_VERSION))/binary>>),
```

So, it is easy enough to set this to something like `Zotonic` or the empty string.

**Comment by mmzeeman** [comment]
On 0.x you will get an empty header.

Also remember that people will still be able to see the site is running on zotonic by looking at the javascript files which are included on the page.

Our client (a bank) was ok with this.

**Comment by mworrell** [comment]
Ok, so it should  be possible to default to an empty 'Server' header.
I propose make it easily configurable (in the zotonic.config) and default to 'Zotonic' ?

**Comment by mworrell** [comment]
One issue remaining, on a redirect we still need to set the server header:

```
Lamma:zotonic-master marc$ curl -I 'http://127.0.0.1:8000/'
HTTP/1.1 302 Found
content-length: 0
date: Thu, 12 Oct 2017 09:07:54 GMT
location: https://127.0.0.1:8443/
server: Cowboy
```

**Comment by mworrell** [comment]
Fixed, see merge request #1824
